### PR TITLE
Enable 2024.5.0

### PR DESCRIPTION
**Pull request by artanokhov**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
 package:
   name: {{ name|lower }}-split
   version: {{ version }}
 
 source:
   - url: https://github.com/openvinotoolkit/openvino/archive/refs/tags/{{ version }}.tar.gz
-    sha256: aeb6297e3e88b4ab7803b27b231a72bd0493b6761ef98f225142146be500f538
+    sha256: 3d5b89760e1c946aca7b219dff4bf4cf5eb511932863c90cafc47d3c01c85744
     patches:
     - patches/001-protobuf-version.patch
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:13]: we need to drop this patch as it's fixed here https://github.com/openvinotoolkit/openvino/commit/79b0baddd05652b88092fccd9cb1c14846ea77a3#diff-a384efac67b1ca1369c48d6daa51cf43dddd459c2d578ce890f1cad632664e92

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,38 +1,37 @@
...
+    sha256: 1e7b4140eed15aec2be36075a322e352013292b28e842608476e9d1ce3b77959  # [linux64]
     folder: src/plugins/intel_npu/thirdparty/level-zero-ext  # [linux64]
-  - url: https://github.com/onnx/onnx/archive/refs/tags/v1.16.0.tar.gz
-    sha256: 0ce153e26ce2c00afca01c331a447d86fbf21b166b640551fe04258b4acfc6a4
+  - url: https://github.com/onnx/onnx/archive/refs/tags/v1.17.0.tar.gz
+    sha256: 8d5e983c36037003615e5a02d36b18fc286541bf52de1a78f6cf9f32005a820e
     folder: thirdparty/onnx/onnx
+  - url: https://github.com/oneapi-src/level-zero/archive/fd459f28041c176df23a0b7b791ff20a7689d237.tar.gz  # [linux64]
+    sha256: e078cae698e368c4e8e57d268f4ef0b766ed8f5548ac8e4fc0a07f4c406bb50d  # [linux64]
+    folder: thirdparty/level-zero  # [linux64]
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:34]:
```suggestion
    folder: thirdparty/level-zero/level-zero  # [linux64]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -59,6 +58,7 @@ requirements:
     - ocl-icd  # [linux64]
     - clhpp  # [linux64 or win64]
     - khronos-opencl-icd-loader  # [win64]
+    - yaml-cpp  # [linux64]
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:61]: NPU requires subdirectory https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/thirdparty/CMakeLists.txt#L21

We need to add `find_package` support to NPU thirdparty

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
+    sha256: 1e7b4140eed15aec2be36075a322e352013292b28e842608476e9d1ce3b77959  # [linux64]
     folder: src/plugins/intel_npu/thirdparty/level-zero-ext  # [linux64]
-  - url: https://github.com/onnx/onnx/archive/refs/tags/v1.16.0.tar.gz
-    sha256: 0ce153e26ce2c00afca01c331a447d86fbf21b166b640551fe04258b4acfc6a4
+  - url: https://github.com/onnx/onnx/archive/refs/tags/v1.17.0.tar.gz
+    sha256: 8d5e983c36037003615e5a02d36b18fc286541bf52de1a78f6cf9f32005a820e
     folder: thirdparty/onnx/onnx
+  - url: https://github.com/oneapi-src/level-zero/archive/fd459f28041c176df23a0b7b791ff20a7689d237.tar.gz  # [linux64]
+    sha256: e078cae698e368c4e8e57d268f4ef0b766ed8f5548ac8e4fc0a07f4c406bb50d  # [linux64]
+    folder: thirdparty/level-zero/level-zero  # [linux64]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:34]: I’m not too sure what level-zero is but can we try to use the conda-forge package?

https://github.com/conda-forge/level-zero-feedstock

Comment (ilya-lavrenov) [reply recipe/meta.yaml:34]: Currently - not.

- OpenVINO NPU plugin depends on older version of level-zero. PR to upgrade https://github.com/openvinotoolkit/openvino/pull/27632
- level-zero's pkg-config files are broken and fix is https://github.com/oneapi-src/level-zero/pull/243

File: recipe/meta.yaml [hunk 5]
```diff
@@ -392,6 +392,7 @@ outputs:
         - {{ pin_subpackage('libopenvino', exact=True) }}
         - pugixml
         - tbb-devel
+        - yaml-cpp
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:395]: it's not required as it's used under NPU internal option only

File: recipe/patches/001-protobuf-version.patch
Comment (ilya-lavrenov) [inline recipe/patches/001-protobuf-version.patch:1]: looks like we need another protobuf patch https://github.com/openvinotoolkit/openvino/pull/27510

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

**Comment by artanokhov** [comment]
> Hi! This is the friendly automated conda-forge-linting service.
> 
> I wanted to let you know that I linted all conda-recipes in your PR (`recipe/meta.yaml`) and found some lint.
> 
> Here's what I've got...
> 
> For **recipe/meta.yaml**:
> 
> * ❌ The recipe is not parsable by any of the known recipe parsers (['conda-forge-tick (the bot)', 'conda-recipe-manager', 'conda-souschef (grayskull)']). Please check the logs for more information and ensure your recipe can be parsed.
> 
> For **recipe/meta.yaml**:
> 
> * ℹ️ The recipe is not parsable by parser `conda-forge-tick (the bot)`. Your recipe may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.
> * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. Your recipe may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.
> * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. Your recipe may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.

So I assume we have bug in reporting linter status

At first I didn't found the linter issue in the recipe and decided to rerender and try to refresh the linter status with some commit, and in the logs we have green status
https://github.com/conda-forge/conda-forge-webservices/actions/runs/11939083258

But here still red

@hmaarrfk Could you please take a look?

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by xhochy** [comment]
This can be a real bug, but I don't know where you could find the logs. The error indicates that the @regro-cf-autotick-bot will not be able to propose any PRs to this repository. You can still merge though, but it is generally recommended to simplify the recipe a bit so that it gets parsable by these tools.

**Comment by artanokhov** [comment]
> This can be a real bug, but I don't know where you could find the logs. The error indicates that the @regro-cf-autotick-bot will not be able to propose any PRs to this repository. You can still merge though, but it is generally recommended to simplify the recipe a bit so that it gets parsable by these tools.

I don't think it's valid errors because if you take a look into changes in the meta.yml - they are common(changed hashes for repos, updated deps, updates in maintainers list). 
Is there any possibility to (manually) re-scan the recipe with linters to check what exactly should be tuned/fixed?

**Comment by xhochy** [comment]
> I don't think it's valid errors because if you take a look into changes in the meta.yml 

It's not due to the changes but simply due to new checks added to the linter.

**Comment by artanokhov** [comment]
> It's not due to the changes but simply due to new checks added to the linter.
Could you please clarify - are there any recommendations how to simplify the recipe?
Or maybe description of new requirements to pass the linter?

**Comment by artanokhov** [comment]
> The linter message is currently too simple, ignore it for now and wait until it gets updated to provide more information.
Got it! Thank you!

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

**Comment by xhochy** [comment]
> The linter message is currently too simple, ignore it for now and wait until it gets updated to provide more information.

The linter now includes a link to its run and you can inspect the logs. Now you can look into it whether it is something that is fixable (quite often souschef trips over things that you cannot fix and should be fixed there instead of the recipe)

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

**Comment by artanokhov** [comment]
> The linter now includes a link to its run and you can inspect the logs. Now you can look into it whether it is something that is fixable (quite often souschef trips over things that you cannot fix and should be fixed there instead of the recipe)

So, I see issues now - thanks.
It seems like following construction can't be parsed

```
    test:
      commands:
        {% set openvino_libs = [] %}
        {{ openvino_libs.append("") or "" }}
        {{ openvino_libs.append("_c") or "" }}
        {% for openvino_lib in openvino_libs %}
        - test -f $PREFIX/lib/libopenvino{{ openvino_lib }}${SHLIB_EXT}.{{ version_postfix }}  # [linux]
        - test -f $PREFIX/lib/libopenvino{{ openvino_lib }}.{{ version_postfix }}${SHLIB_EXT}  # [osx]
        - if not exist %LIBRARY_BIN%\\openvino{{ openvino_lib }}%SHLIB_EXT% exit 1  # [win]
        {% endfor %}
```
because of {%%} conditions inside
yaml-linter expect that all commands inside the block will start with `-`

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

**Comment by artanokhov** [comment]
> because of jinja-blocks inside yaml-linter expect that all commands inside the block will start with `-`

I've removed the blocks but the linter doesn't rerun or the comment doesn't updated properly with new link
FYI @xhochy

**Comment by xhochy** [comment]
It actually looks like it run: https://github.com/conda-forge/conda-forge-webservices/actions/runs/12069274925/job/33656184032

**Comment by xhochy** [comment]
The comment is auto-updating

**Comment by artanokhov** [comment]
So, currently linter is green but in the logs I see that probably there is an error in the `conda-recipe-manager` parser itself
```
Traceback (most recent call last):
  File "/opt/conda/envs/cf-feedstock-ops/lib/python3.11/site-packages/conda_smithy/linter/lints.py", line 1069, in lint_recipe_is_parsable
    RecipeParser(recipe_text)
  File "/opt/conda/envs/cf-feedstock-ops/lib/python3.11/site-packages/conda_recipe_manager/parser/recipe_reader.py", line 4[83](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12069683030/job/33657483944#step:5:85), in __init__
    parent = node_stack[-1]
             ~~~~~~~~~~^^^^
IndexError: list index out of range
```

**Comment by xhochy** [comment]
> currently linter is green but in the logs

The run will always pass so that it can post the message here.

**Pull request merged**